src/Microsoft.VisualBasic.Core/src/Microsoft/VisualBasic/MyServices/Internal/ContextValue.vb

```diff
@@ -0,0 +1,63 @@
+' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Option Explicit On
+Option Strict On
+
+Imports System.Collections
+Imports System.Collections.Generic
+Imports System.Threading
+
+Namespace Microsoft.VisualBasic.MyServices.Internal
+
+    '''**************************************************************************
+    ''' ;ContextValue
+    ''' <summary>
+    ''' Stores an object in a context appropriate for the environment we are 
+    ''' running in (web/windows)
+    ''' </summary>
+    ''' <typeparam name="T"></typeparam>
+    ''' <remarks>
+    ''' "Thread appropriate" means that if we are running on ASP.Net the object will be stored in the 
+    ''' context of the current request (meaning the object is stored per request on the web).
+    ''' Note that an instance of this class can only be associated
+    ''' with the one item to be stored/retrieved at a time.
+    ''' </remarks>
+    <Global.System.ComponentModel.EditorBrowsableAttribute(Global.System.ComponentModel.EditorBrowsableState.Never)>
+    Public Class ContextValue(Of T)
+        Public Sub New()
+            m_ContextKey = System.Guid.NewGuid.ToString
+        End Sub
+
+        '''**************************************************************************
+        ''' ;Value
+        ''' <summary>
+        ''' Get the object from the correct thread-appropriate location
+        ''' </summary>
+        ''' <value></value>
+        Public Property Value() As T 'No Synclocks required because we are operating upon instance data and the object is not shared across threads
+            Get
+                Dim dictionary As IDictionary = GetDictionary()
+                Return DirectCast(dictionary(m_ContextKey), T) 'Note, IDictionary(key) can return Nothing and that's ok
+            End Get
+            Set(ByVal value As T)
+                Dim dictionary As IDictionary = GetDictionary()
+                dictionary(m_ContextKey) = value
```

Comment:
This seems like it could be a leak, e.g.
```C#
var cv = new ContextValue<object>();
cv.Value = new object();
```
the underlying dictionary will, which will contain an entry for this Guid/object pair, will end up staying around as long as this thread stays around, even once `cv` is collected and long gone.

I have no sense for how this `ContextValue<T>` class is used, and given that the only way to construct one is hidden with EditorBrowsableState.Never and its in an Internal namespace (and that I never heard of it until today) makes me think it's not particularly important and this might be a fine result.  But wanted to mention it in case it's not.